### PR TITLE
documentation for offsetting

### PR DESCRIPTION
**Pull request by Wikunia**
Start of documentation for offsetting a polygon and creating polygons with `IntPoint`.
Missing:
- link to documentation in readme
- polygon clipping

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+```
+
+and back
+```julia
+tofloat.(ips, 1, 3)
+```
+
+## Offsetting a polygon
+
+A polygon can be offsetted (inflated/deflated) in the following way using Clipper.jl. More information about general offsetting in the [Clipper documentation](http://www.angusj.com/delphi/clipper/documentation/Docs/Units/ClipperLib/Classes/ClipperOffset/_Body.htm).
```
Comment (sjkelly) [inline docs/src/index.md]:
```suggestion
[Original Clipper documentation](http://www.angusj.com/delphi/clipper/documentation/Docs/Units/ClipperLib/Classes/ClipperOffset/_Body.htm).

A polygon can be offset (inflated/deflated) in the following way using Clipper.jl:
```

File: docs/src/index.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+miter_offset_polygons = execute(co, 7.0)
+```
+
+![Different offsetting types](assets/offset_diff_types.png)
+
+`JoinTypeRound` produces rounded corners, `JoinTypeSquare` produces squared corners.
+In the case of `JoinTypeMiter` it depends on the degree of the angle or in different words the maximum offsetted distance if corners.
+would not be squared. If the maximum distance is bigger than `MitterLimit * delta` than it is squared which would be the case in the upper right corner.
+
+The `MitterLimit` is the first argument of `ClipperOffset` for example `ClipperOffset(3.0)` would set it to `3.0` (default is `2.0`).
```
Comment (sjkelly) [inline docs/src/index.md]:
```suggestion
The `MiterLimit` is the first argument of `ClipperOffset` for example `ClipperOffset(3.0)` would set it to `3.0` (default is `2.0`).
```

File: docs/src/index.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+co = ClipperOffset()
+add_path!(co, polygon, JoinTypeMiter, EndTypeClosedPolygon)
+miter_offset_polygons = execute(co, 7.0)
+```
+
+![Different offsetting types](assets/offset_diff_types.png)
+
+`JoinTypeRound` produces rounded corners, `JoinTypeSquare` produces squared corners.
+In the case of `JoinTypeMiter` it depends on the degree of the angle or in different words the maximum offsetted distance if corners.
+would not be squared. If the maximum distance is bigger than `MitterLimit * delta` than it is squared which would be the case in the upper right corner.
```
Comment (sjkelly) [inline docs/src/index.md]:
```suggestion
would not be squared. If the maximum distance is bigger than `MiterLimit * delta` than it is squared which would be the case in the upper right corner.
```

**Review by sjkelly — APPROVED** [review]
Looks great, thanks!

File: docs/src/index.md (hunk 2 above)
Comment (Wikunia) [reply docs/src/index.md]: Oh good catch 😅

**Comment by sjkelly** [comment]
Sweet, lets see how the deploy goes :) Thanks again.

**Comment by Wikunia** [comment]
You're welcome. Is gh-pages activated for this repo?

**Comment by sjkelly** [comment]
Up now, looks great: https://juliageometry.github.io/Clipper.jl/dev/

**Pull request merged**